File: OpenWeather/Utils/SelectAllTextBehavior.vb
Imports System.Windows.Interactivity

Public Class SelectAllTextBehavior
    Inherits Behavior(Of TextBox)

    Protected Overrides Sub OnAttached()
        MyBase.OnAttached()
        AddHandler AssociatedObject.PreviewKeyUp, AddressOf AssociatedObject_PreviewKeyUp
    End Sub

    Protected Overrides Sub OnDetaching()
        MyBase.OnDetaching()
        RemoveHandler AssociatedObject.PreviewKeyUp, AddressOf AssociatedObject_PreviewKeyUp
    End Sub

    Private Sub AssociatedObject_PreviewKeyUp(ByVal sender As Object, ByVal e As KeyEventArgs)
        If (e.Key = Key.Enter) Then
            AssociatedObject.SelectAll()
        End If
    End Sub
End Class

File: OpenWeatherCS/Utils/SelectAllTextBehavior.cs
using System.Windows.Controls;
using System.Windows.Input;
using System.Windows.Interactivity;

namespace OpenWeatherCS.Utils
{
    class SelectAllTextBehavior : Behavior<TextBox>
    {
        protected override void OnAttached()
        {
            base.OnAttached();
            AssociatedObject.PreviewKeyUp += AssociatedObject_PreviewKeyUp;
        }

        protected override void OnDetaching()
        {
            base.OnDetaching();
            AssociatedObject.PreviewKeyUp -= AssociatedObject_PreviewKeyUp;
        }

        private void AssociatedObject_PreviewKeyUp(object sender, KeyEventArgs e)
        {
            if (e.Key == Key.Enter)
            {
                AssociatedObject.SelectAll();
            }
        }
    }
}
